benchmarks/bench_plot_onnxruntime_logreg.py

```diff
@@ -0,0 +1,230 @@
+
+
+##############################
+# Implementations to benchmark.
+##############################
+
+def fcts_model(X, y, fit_intercept):
+    "LogisticRegression."
+    rf = LogisticRegression(fit_intercept=fit_intercept)
+    rf.fit(X, y)
+
+    initial_types = [('X', FloatTensorType([1, X.shape[1]]))]
+    onx = convert_sklearn(rf, initial_types=initial_types)
+    f = BytesIO()
+    f.write(onx.SerializeToString())
+    content = f.getvalue()
+    sess = InferenceSession(content)
+
+    outputs = [o.name for o in sess.get_outputs()]
+
+    def predict_skl_predict(X, model=rf):
+        return rf.predict(X)
+
+    def predict_skl_predict_proba(X, model=rf):
+        return rf.predict_proba(X)
+
+    def predict_onnxrt_predict(X, sess=sess):
+        return numpy.array(sess.run(outputs[:1], {'X': X.astype(np.float32)}))
+
+    def predict_onnxrt_predict_proba(X, sess=sess):
+        res = sess.run(outputs[1:], {'X': X.astype(np.float32)})[0]
+        # do not use DataFrame to convert the output into array,
+        # it takes too much time
+        out = numpy.empty((len(res), len(res[0])), dtype=numpy.float32)
+        for i, row in enumerate(res):
+            for k, v in row.items():
+                out[i, k] = v
+        return out
+
+    return {'predict': (predict_skl_predict,
+                        predict_onnxrt_predict),
+            'predict_proba': (predict_skl_predict_proba,
+                              predict_onnxrt_predict_proba)}
+
+
+##############################
+# Benchmarks
+##############################
+
+def allow_configuration(**kwargs):
+    return True
+
+
+def bench(n_obs, n_features, fit_intercepts, methods,
+          repeat=10, verbose=False):
+    res = []
+    for nfeat in n_features:
+
+        ntrain = 100000
+        X_train = np.empty((ntrain, nfeat))
+        X_train[:, :] = rand(ntrain, nfeat)[:, :]
+        X_trainsum = X_train.sum(axis=1)
+        eps = rand(ntrain) - 0.5
+        X_trainsum_ = X_trainsum + eps
+        y_train = (X_trainsum_ >= X_trainsum).ravel().astype(int)
+
+        for fit_intercept in fit_intercepts:
+            fcts = fcts_model(X_train, y_train, fit_intercept)
+
+            for n in n_obs:
+                for method in methods:
+
+                    fct1, fct2 = fcts[method]
+
+                    if not allow_configuration(n=n, nfeat=nfeat, fit_intercept=fit_intercept):
+                        continue
+
+                    obs = dict(n_obs=n, nfeat=nfeat,
+                               fit_intercept=fit_intercept, method=method)
+
+                    # creates different inputs to avoid caching in any ways
+                    Xs = []
+                    for r in range(repeat):
+                        x = np.empty((n, nfeat))
+                        x[:, :] = rand(n, nfeat)[:, :]
+                        Xs.append(x)
+
+                    # measures the baseline
+                    st = time()
+                    r = 0
```

Comment:
Also, what is r? Could the names be a little more descriptive?

Reply:
renamed into repeated